src/main/java/com/o19s/es/ltr/feature/FeatureSet.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright [2017] Wikimedia Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.o19s.es.ltr.feature;
+
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * A set of features.
+ * Features can be identified by their name or ordinal.
+ * A FeatureSet is dense and ordinals start at 0.
+ * Time critical processes will most likely use ordinals so that they can
+ * represent the set as a flat Feature[] array.
+ */
+public interface FeatureSet extends Iterable<Feature> {
+
+    /**
+     * Name of the feature set
+     * @return the name
+     */
+    String name();
+
+    /**
+     * List of feature present in this model.
+     * This list must be properly ordered to that {@link #featureOrdinal(String)}
+     * matches features().get(ordinal)
+     * @return the list of features in the set.
+     */
+    List<Feature> features();
```

Comment:
right, reading back this interface I'm not super happy with it. I'll simplify a bit.
I'm anticipating some future usage and it's likely to change when I'll have to implement real stuff on top of it.